pkg/kubectl/resource_printer.go

```diff
@@ -2256,9 +2265,78 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 		}
 		return resultValue.Interface().(error)
 	}
+
+	// we don't recognize this type, but we can still attempt to print some reasonable information about.
+	unstructured, ok := obj.(*runtime.Unstructured)
```

Comment:
Keep in mind unstructured really is "generic API v1" when used this way (with accessors).  So this code only works until someone changes metadata and then is horribly broken again.  Which is not inconsistent with what we've said up till now, that we won't change v1 metadata until we have server side transformation (we can't, effectively).